tsl/src/continuous_aggs/utils.c

```diff
@@ -29,31 +17,37 @@ enum
 	_Anum_cagg_validate_query_max
 };
 
+#define Natts_cagg_validate_query (_Anum_cagg_validate_query_max - 1)
+
 static Datum
 create_cagg_validate_query_datum(TupleDesc tupdesc, const bool is_valid_query,
 								 const ErrorData *edata)
 {
-	Datum values[_Anum_cagg_validate_query_max] = { 0 };
-	bool nulls[_Anum_cagg_validate_query_max] = { false };
+	NullableDatum datums[Natts_cagg_validate_query] = { 0 };
 	HeapTuple tuple;
 
 	tupdesc = BlessTupleDesc(tupdesc);
 
-	values[AttrNumberGetAttrOffset(Anum_cagg_validate_query_valid)] = BoolGetDatum(is_valid_query);
-
-	fill_values_text(Anum_cagg_validate_query_error_level,
-					 values,
-					 nulls,
-					 edata->elevel > 0 ? (char *) error_severity(edata->elevel) : NULL);
-	fill_values_text(Anum_cagg_validate_query_error_code,
-					 values,
-					 nulls,
-					 edata->sqlerrcode > 0 ? unpack_sql_state(edata->sqlerrcode) : NULL);
-	fill_values_text(Anum_cagg_validate_query_error_message, values, nulls, edata->message);
-	fill_values_text(Anum_cagg_validate_query_error_detail, values, nulls, edata->detail);
-	fill_values_text(Anum_cagg_validate_query_error_hint, values, nulls, edata->hint);
-
-	tuple = heap_form_tuple(tupdesc, values, nulls);
+	ts_datum_set_bool(Anum_cagg_validate_query_valid, datums, is_valid_query);
+	ts_datum_set_text(Anum_cagg_validate_query_error_level,
```

Comment:
What do you think about introducing a `ts_datum_set_cstring` function as well which calls `cstring_to_text` internally?

Reply:
Or leave it as is but instead of receive a `text *` the function receive a `char *` and then internally we call the `cstring_to_text`. WDYT?

Reply:
Created a `ts_datum_set_text_from_cstring` that internally call the `cstring_to_text`... in the future we can introduce a simple `ts_datum_set_text` that receive a `text *` if necessary!